pkg/controller/plan/migration.go

```diff
@@ -1638,13 +1648,27 @@ func (r *Migration) updatePopulatorCopyProgress(vm *plan.VMStatus, step *plan.St
 	return
 }
 
+func (r *Migration) isPopulatorFailed(populatorPods []core.Pod, givenPvcId string) (bool, error) {
+	for _, pod := range populatorPods {
+		pvcId := pod.Name[len(PopulatorPodPrefix):]
+		if givenPvcId != pvcId {
+			continue
+		}
+		if pod.Status.Phase == core.PodFailed {
+			return true, fmt.Errorf("populator pod %s/%s failed for PVC %s. Please check the pod logs.", pod.Namespace, pod.Name, pvcId)
```

Comment:
if we keep this function, it's not that good to return both boolean and error here since an error that is returned from a function that is called isPopulatorFailed suggests that when an error is returned, it means we **couldn't check** whether the populator pod failed..
I'd change this to return just an error when this function return true if needed